cs/src/Contracts/ClusterDetails.cs

```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.VsSaaS.TunnelService.Contracts;
+
+/// <summary>
+/// Tunnel service cluster details.
+/// </summary>
+public class ClusterDetails {
+    /// <summary>
+    /// A cluster identifier based on its region.
+    /// </summary>
+    public string? ClusterId { get; set; }
```

Comment:
Why is it nullable? It should always have a value.

(In some other contracts, some required properties are nullable to support partial updates. But there will not be an API for updating clusters.)